WarehouseManagement.Api/Controllers/ZoneController.cs

```diff
@@ -86,5 +88,39 @@ public async Task<IActionResult> AllDeleted()
 
             return Ok(model);
         }
+
+        [HttpGet("entries")]
+        [ProducesResponseType(200)]
+        [ProducesResponseType(404)]
+        public async Task<IActionResult> Entries(int zoneId, [FromQuery] string[] statuses)
+        {
+            var predicates = new List<Predicate<Entry>>();
+
+            if (statuses.Contains(Waiting))
+            {
+                predicates.Add(entry =>
+                    entry.StartedProccessing == null && entry.FinishedProccessing == null
+                );
+            }
+
+            if (statuses.Contains(Finished))
+            {
+                predicates.Add(entry => entry.FinishedProccessing != null);
+            }
+
+            if (statuses.Contains(Processing))
+            {
+                predicates.Add(entry =>
+                    entry.StartedProccessing != null && entry.FinishedProccessing == null
+                );
+            }
+
+            Predicate<Entry> combinedPredicate = entry =>
+                predicates.Any(predicate => predicate(entry));
```

Comment:
Maybe have this into its own method inside the service, a util function or inside `ZoneEntryStatuses` classx?

Also i think it might make sense to have for each method you add to `predicates` to be its own method too, for example `GetWaitingEntries`, not 100% sure about this, but I think the above point makes sense, you can think about this one.

Reply:
Makes sense, I will refactor it so that the service can provide `GetWaitingEntries`, `GetProccessingEntries` and `GetFinishedEntries` methods.

Reply:
Let me know if you want separate actions (endpoints) for each status. Right now the statuses are extracted from the query parameter passed in the URL, when accessing `api/zone/entries`.